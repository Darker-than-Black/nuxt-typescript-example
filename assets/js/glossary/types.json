{
  "String": "string",
  "Array": "array",
  "Number": "number",
  "Object": "object"
}
