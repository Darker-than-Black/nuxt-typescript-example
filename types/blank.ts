export interface BlankItemField {
  title: string,
  path: string,
  itemClass?: string,
  countryPath?: string
}

export interface BlankItemList extends BlankItemField {
  items: string[]
}

export interface BlankItemConfig {
  type: string,
  path: string,
  title?: string,
  countryPath?: string,
  width?: string,
  class?: string,
  headerClass?: string,
  itemsContentClass?: string,
  itemListClass?: string,
  items: BlankItemField[],
  icons?: BlankItemList[]
}
