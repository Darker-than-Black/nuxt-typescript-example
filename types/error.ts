export type HttpError = {
  field: string,
  message: string
}

export type ErrorHandler = {
  statuses: Array<number>,
  return: any
}

export type ErrorConfig = {
  returnValue?: any,
  errorsList?: Array<ErrorHandler>
}

export type ErrorCode = {
  error?: number
}
