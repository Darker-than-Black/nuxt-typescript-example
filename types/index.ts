export * from './input'
export * from './router'
export * from './error'
export * from './table'
export * from './link'
export * from './blank'

// api
export * from './api/declaration/declarations'
export * from './api/declaration/declaration'
export * from './api/car-goods'
export * from './api/entry-journal/entry-journal'
export * from './api/entry-journal/entry-journal-ticket'
export * from './api/dictionary'

// UI
export * from './UI'

// tests
export * from './unit-test'
