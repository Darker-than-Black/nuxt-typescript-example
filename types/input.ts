export type InputValue = number|string|boolean

export type MultiSelectValue = InputValue[]

export type SelectOptions = SelectOption[]

export type SelectOption = {
  value: InputValue,
  text: string
}

export type TableRowValue = {
  [key: string]: FormFieldValue
}

export type FormFieldValue = InputValue|MultiSelectValue[]|TableRowValue[]|object

export interface IFormFieldComponentProps {
  value?: FormFieldValue,
  readonly label?: string,
  readonly disabled?: boolean,
  readonly inputClasses?: string[]|string,
  // input attrs
  readonly type?: string,
  readonly placeholder?: string,
  // select attrs
  readonly options?: SelectOptions,
  readonly filterKey?: string,
  // for validate
  readonly vid?: string,
  readonly rules?: string,
  // for data list
  readonly listLink?: string,
}

export interface IFactoryFieldConfig extends IFormFieldComponentProps {
  // ~/assets/js/glossary/field-types.json
  readonly fieldType: string,
  readonly defaultValue: FormFieldValue,
  readonly invisible?: boolean,
  readonly fields?: {
    [key: string]: IFactoryFieldConfig
  }
}

export interface IFactoryFieldsConfig {
  [key: string]: IFactoryFieldConfig
}
