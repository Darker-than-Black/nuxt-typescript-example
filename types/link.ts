export type Links = {
  [key: string]: Link
}

export type Link = {
  text: string,
  icon?: string,
  path: string,
  subLinks?: Links
}
