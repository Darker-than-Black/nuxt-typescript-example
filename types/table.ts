export type TableConfig = {
  size?: number,
  totalElements?: number,
  totalPages?: number,
  number?: number // page number
}

// Doc: https://bootstrap-vue.org/docs/components/table#field-definition-reference
export type TableColConfig = {
  key: string,
  label: string,
  sortable: boolean,
  tdClass: string[] | string,
  thClass: string[] | string
}

export type TableSortEvent = {
  apiUrl: string,
  currentPage: number,
  filter: any
  perPage: number,
  sortBy: string
  sortDesc: boolean
}
