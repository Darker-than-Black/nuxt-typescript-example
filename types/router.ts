import { InputValue, MultiSelectValue } from '~/types/input'

export type QueryParam = MultiSelectValue | InputValue

export type QueryParams = {
  [key: string]: QueryParam
}

export type QueryParamsConfig = {
  [key: string]: {
    defaultValue: QueryParam,
    type: string
  }
}
