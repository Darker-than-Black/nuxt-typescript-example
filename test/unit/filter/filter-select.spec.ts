import vClickOutside from 'v-click-outside'
import VueTestInstance from '~/test/factory/vue-instance'
import FilterSelect from '~/components/filter/filter-select.vue'
import { SelectOption } from '~/types'

describe('filter-select.vue', () => {
  let wrapper: any
  const options: SelectOption[] = [
    { value: 'val_1', text: 'Val 1' },
    { value: 'val_2', text: 'Val 2' }
  ]

  beforeEach(() => {
    wrapper = VueTestInstance(FilterSelect, {
      isShallow: false,
      modules: [vClickOutside],
      options: {
        propsData: {
          showOptions: true,
          label: 'Filter Select',
          value: '',
          options
        }
      }
    })
  })

  it('initialize correctly', () => {
    expect(wrapper.vm).toBeTruthy()
  })

  it('should match the snapshot', () => {
    expect(wrapper.html()).toMatchSnapshot()
  })

  it('check emit input value', () => {
    wrapper.findAll('li.filter__item').at(0).trigger('click')
    expect(wrapper.emitted().input).toEqual([[options[0].value, options[0].value]])
  })

  it('check input function', () => {
    expect(wrapper.vm.input(options[1].value)).toBe(options[1].value)
  })

  it('check input function, when input value === prop value', () => {
    wrapper.setProps({ value: options[1].value })
    expect(wrapper.vm.input(options[1].value)).toBe('')
  })
})
