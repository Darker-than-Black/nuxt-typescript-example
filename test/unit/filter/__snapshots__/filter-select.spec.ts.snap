// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`filter-select.vue should match the snapshot 1`] = `
<div class="filter__wrapper" showoptions="true"><label for="Filter Select" class="title__field">
    Filter Select
    <!----></label>
  <div class="filter">
    <p class="filter__label">
      Оберіть зі списку
      <img src="/icons/Arrow-down.svg" alt="arrow" class="img filter__icon"></p>
    <ul class="filter__list" style="display: none;">
      <li class="filter__item">
        Val 1
      </li>
      <li class="filter__item">
        Val 2
      </li>
    </ul>
  </div>
</div>
`;
